src/Compilers/CSharp/Test/Emit3/FieldKeywordTests.cs

```diff
@@ -12131,5 +12131,512 @@ public string Prop
+        }
+
+        [Fact]
+        public void Nullable_Accessor_FieldStateAffectedByAssigningThis()
+        {
+            var source = """
+                #nullable enable
+
+                public struct S
+                {
+                    public string Prop
+                    {
+                        get
+                        {
+                            field = "a";
+                            this = default;
+                            field.ToString(); // 1
+                            return field;
+                        }
+                        set
+                        {
+                            field = "a";
+                            this = default;
+                            field.ToString(); // 2
+                        }
+                    }
+                }
+                """;
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics(
+                // (11,13): warning CS8602: Dereference of a possibly null reference.
+                //             field.ToString(); // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "field").WithLocation(11, 13),
+                // (18,13): warning CS8602: Dereference of a possibly null reference.
+                //             field.ToString(); // 2
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "field").WithLocation(18, 13));
+        }
+
+        [Fact]
+        public void Nullable_Constructor_FieldStateAffectedByAssigningThis()
+        {
+            var source = """
+                #nullable enable
+
+                public struct S
+                {
+                    public string Prop
+                    {
+                        get => field;
+                        set => field = value;
+                    }
+
+                    public static string StaticProp
+                    {
+                        get => field;
+                        set => field = value;
+                    } = "b";
+
+                    public S(string str)
+                    {
+                        Prop = str;
+                    }
+
+                    public S() : this("a")
+                    {
+                        Prop.ToString();
+                        this = default;
+                        Prop.ToString(); // 1
+                        StaticProp.ToString();
+                    }
+                }
+                """;
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics(
+                // (26,9): warning CS8602: Dereference of a possibly null reference.
+                //         Prop.ToString(); // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "Prop").WithLocation(26, 9));
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/77991")]
+        public void Nullable_OrdinaryMethod_FieldStateAffectedByAssigningStruct()
```

Comment:
>Nullable_OrdinaryMethod_FieldStateAffectedByAssigningStruct [](http://example.com/codeflow?start=20&length=59)

Is this test or the next two tests affected by the `NullableWalker` change?

Reply:
No, they are not. These are here to illustrate a more general problem with nullable state management in structs interacting with field-backed prop slot sharing.